### ISSUE TITLE
VSCode main icon not showing (Linux/deb)

### ISSUE THREAD

**Issue by zaid-khan** [microsoft/vscode#71942]
Issue Type: <b>Bug</b>

Try to open from the start menu or from anywhere else for that matter. The main visual studio icon isn't present.  Problem started with the latest March update.

VS Code version: Code 1.33.0 (0dd516dd412d42323fc3464531b1c715d51c4c1a, 2019-04-04T15:08:57.775Z)
OS version: Linux x64 4.15.0-46-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz (4 x 2327)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>surface_synchronization: enabled_on<br>video_decode: unavailable_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|5, 3, 1|
|Memory (System)|15.44GB (12.56GB free)|
|Process Argv|--unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-django|bat|0.19.0
vsc-material-theme|Equ|2.8.2
prettier-vscode|esb|1.8.1
python|ms-|2019.3.6352
cpptools|ms-|0.22.1
csharp|ms-|1.18.0
material-icon-theme|PKi|3.7.0
java|red|0.42.1
vscodeintellicode|Vis|1.1.5
vscode-java-debug|vsc|0.17.0
vscode-java-dependency|vsc|0.4.0
vscode-java-pack|vsc|0.6.0
vscode-java-test|vsc|0.15.1
vscode-maven|vsc|0.16.0

</details>

**Comment by zaid-khan** [comment]
Have a look at the start menu and the bottom taskbar: ![Screenshot from 2019-04-08 20-01-59](https://user-images.githubusercontent.com/17180221/55732408-8e630300-5a39-11e9-8bf6-8b85945615aa.png)

**Comment by epsniff** [comment]
Seeing the same issue on Ubuntu 18.04.2 LTS:

<details><summary>Screen Shot</summary>
<img src="https://user-images.githubusercontent.com/2811550/55740849-d0138f80-59e0-11e9-9672-8900b361e207.png">
</details>

**Comment by Tyriar** [comment]
Can someone seeing this issue post the output of this?

```bash
cat /usr/share/applications/code.desktop
/usr/share/pixmaps | grep visualstudio
```

**Comment by zaid-khan** [comment]
```
[Desktop Entry]
Name=Visual Studio Code
Comment=Code Editing. Redefined.
GenericName=Text Editor
Exec=/usr/share/code/code --unity-launch %F
Icon=com.visualstudio.code
Type=Application
StartupNotify=false
StartupWMClass=Code
Categories=Utility;TextEditor;Development;IDE;
MimeType=text/plain;inode/directory;
Actions=new-empty-window;
Keywords=vscode;

X-Desktop-File-Install-Version=0.23

[Desktop Action new-empty-window]
Name=New Empty Window
Exec=/usr/share/code/code --new-window %F
Icon=com.visualstudio.code
```

The second command gives an error:

`bash: /usr/share/pixmaps: Is a directory`

**Comment by Tyriar** [comment]
@zaid-khan doh, I meant `ls /usr/share/pixmaps | grep visualstudio` for the second command

**Comment by zaid-khan** [comment]
`com.visualstudio.code.png`

Also, I updated to `1.33.1` and the issue still persists.

**Comment by machinebiologist** [comment]
> Can someone seeing this issue post the output of this?
> 
> ```shell
> cat /usr/share/applications/code.desktop
> /usr/share/pixmaps | grep visualstudio
> ```

```shell
$ ls /usr/share/pixmaps | grep visualstudio
com.visualstudio.code.png
```

**Comment by machinebiologist** [comment]
This worked for me on several machines:

Modify /usr/share/applications/code.desktop

Change:
``` Icon=com.visualstudio.code ```
to
``` Icon=/usr/share/pixmaps/com.visualstudio.code.png ```

**Comment by Tyriar** [comment]
@machinebiologist it's strange that it works on some machines but not others though, it was reported above that it happens on Ubuntu 18.04 but it definitely doesn't for me 🤔

**Comment by machinebiologist** [comment]
> @machinebiologist it's strange that it works on some machines but not others though, it was reported above that it happens on Ubuntu 18.04 but it definitely doesn't for me thinking

@Tyriar: indeed! I have had it occur on two machines running Ubuntu MATE 18.04. Perhaps it's related to the distribution or the desktop environment?

**Comment by zaid-khan** [comment]
Strangely, on my other machine running Linux Mint 19.1 xfce, the desktop icon has changed into some generic one but the menu and taskbar (thankfully) retains the correct old one. 
![Screenshot_2019-04-15_20-45-33](https://user-images.githubusercontent.com/17180221/56144353-dc3eb480-5fbf-11e9-9051-5031fbc245d7.png)

**Comment by machinebiologist** [comment]
@zaid-khan Related behaviour on MATE: the "start menu" showed the proper icon, while the taskbar did not.

**Comment by efremropelato** [comment]
> The following workaround fixed the issue for me on several machines:
> 
> Modify /usr/share/applications/code.desktop
> 
> change:
> `Icon=com.visualstudio.code`
> to
> `Icon=/usr/share/pixmaps/com.visualstudio.code.png`

**Work on Ubuntu 18.04**

**Comment by epsniff** [comment]
The work around worked for me as well, but I also got it working without using the work around.  

I removing and installing vscode many times with the no luck.  But finally instead of using the dep package from the website, I switched to installing it using the `Ubuntu Software`'s market place.  The icon has now reappeared. 

This means the desktop file moves from `/usr/share/applications/code.desktop` to `/var/lib/snapd/desktop/applications/code_code.desktop`.
